src/scripts/dot-gov.js

```diff
@@ -0,0 +1,366 @@
+      filteredDomains = [
+        ...filterByField(DATA_FIELDS.NAME),
+        ...filterByField(DATA_FIELDS.ORG),
+        ...filterByField(DATA_FIELDS.AGENCY),
+        ...filterByField(DATA_FIELDS.CITY),
+        ...filterByField(DATA_FIELDS.STATE),
+        ...filterByField(DATA_FIELDS.TYPE),
+      ];
+      const seen = filteredDomains.map(o => o[DATA_FIELDS.NAME]);
+      filteredDomains = filteredDomains.filter(
+        ({[DATA_FIELDS.NAME]: value}, index) => !seen.includes(value, index + 1)
+      )
+    }
+
+    const numResults = filteredDomains.length;
+
+    // pluralize
+    const domainText = entityExists ? `${entity} domains` : 'domains';
+
+    // format a salutation
+    if (numResults > 0 && searchTermExists) {
+      headerText = `Here's what I could find for ${domainText} related to "${searchTerm}".`
+      const numToDisplay = numResults > 5 ? 5 : numResults;
+      output = selectDomainsAtRandom(filteredDomains, numToDisplay);
+    } else if (numResults > 0) {
+      headerText = `There are ${numResults} \`.gov\` ${domainText} right now! Have you seen this one?`
+      output = selectDomainsAtRandom(filteredDomains, 1);
+    } else if (numResults === 0 && searchTermExists) {
+      headerText = `I found nothing related to "${searchTerm}", sorry.`
+    }
+
+    // add salutation to the response
+    if (headerText) {
+      blocks.push({
+        "type": "section",
+        "text": {
+          "text": headerText,
+          "type": "mrkdwn"
+        }
+      })
+    }
+
+    // get the status of domains
+    const domainStatus = output.map(d => checkDomainStatus(d));
+    await Promise.all(domainStatus);
+
+    // add each domain (max of 5) to the response
+    for (const [index, domain] of output.entries()) {
+      const name = domain[DATA_FIELDS.NAME];
+      const status = domainStatus[index];
+      const agency = domain[DATA_FIELDS.AGENCY];
+      const city = domain[DATA_FIELDS.CITY];
+      const state = domain[DATA_FIELDS.STATE];
+      blocks.push({
+        "type": "section",
+        "text": {
+          "text": `<https://${name}|${name}> (${status}), ` +
+                  `presented by _${agency}_ in _${city}_, _${state}_`,
+          "type": "mrkdwn"
+        }
+      })
+    }
+
+    // let the user know if their search turned up more than 5 results
+    if (numResults > 5 && searchTermExists) {
+      const numOtherResults = numResults - 5;
+      const others = numOtherResults > 1 ? 'others' : 'other';
+      blocks.push({
+        "type": "section",
+        "text": {
+          "text": `. . . plus ${numOtherResults} ${others}.`,
+          "type": "mrkdwn"
+        }
+      })
+    }
+
+    // if there is anything to say, say it!
+    if (blocks.length) {
+      say({
+        icon_emoji: ":dotgov:",
+        thread_ts: thread,
+        username: ".Gov",
+        unfurl_links: false,
+        unfurl_media: false,
+        blocks
+      });
```

Comment:
This will need a `text` property as well for accessibility purposes. Slack isn't the best at a11y. Blocks don't show up in screen readers (or notifications), so the `text` property is used to announce the content for those. It's a hassle.

Reply:
Done!